bionic/executor.py

```diff
@@ -6,19 +6,49 @@
 """
 
 import copy
+from functools import partial
 import logging
+from multiprocessing.managers import SyncManager
 import queue
 import sys
 import threading
 import traceback
+from uuid import uuid4
 
-from multiprocessing.managers import SyncManager
-
+from .aip.task import Task
 from .deps.optdep import import_optional_dependency
 from .utils.misc import oneline, SynchronizedSet
 
 
-class Executor:
+class AipExecutor:
+    """
+    Encapsulates all objects related to remote execution on Google AIP
+    in one place.
+    """
+
+    def __init__(self, aip_job):
+        self._aip_job = aip_job
+
+    def submit(self, fn, *args, **kwargs):
+        assert self._resource is not None
+        return Task(
+            # TODO: Use a better identifiable name, maybe a combination
+            # of entity name and case key.
+            # This is a temporary workaround. We are changing the random
+            # UUID in such a manner because AIP names have to start with
+            # a letter and only accepts alphanumeric and underscore
+            # characters.
+            name="a" + str(uuid4()).replace("-", ""),
+            job=self._aip_job,
+            resource=self._resource,
+            function=partial(fn, *args, **kwargs),
+        ).submit()
+
+    def set_resource(self, resource):
```

Comment:
If we're using different interfaces for the two executors now, does it still make sense to have a separate `set_resource` method?